src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1174,6 +1256,42 @@ LRESULT CALLBACK WindowImplWin32::globalOnEvent(HWND handle, UINT message, WPARA
     return DefWindowProcW(handle, message, wParam, lParam);
 }
 
+////////////////////////////////////////////////////////////
+void WindowImplWin32::prepareTouch()
+{
+    static bool prepared = false;
+    if (!prepared)
+    {
+        prepared = true;
+
+        // Reset touch IDs
+        std::fill_n(touchIDs, maxTouchPoints, static_cast<DWORD>(-1));
```

Comment:
This is one place where using `std::array` would help. We could simply use `std::array::fill`.

Reply:
If we end up using `std::optional`, the `std::array` could just be default-initialized. Another elegant plus 🙂